### PR TITLE
Improve integer arithmetic tests

### PR DESCRIPTION
**Pull request by Akuli**
Now that we have more integer types (#864), we have more corner cases to test.

### CODE REVIEW

File: tests/should_succeed/compare_signed_unsigned.jou [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    printf("%d %d %d\n", a != bb, b != bb, c != bb)    # Output: 1 0 1
+
+    # Numbers aren't necessarily equal if they have the same bits.
+    x = -100 as int8
+    y = 156 as byte
+
+    # Output: same bits 1 1
+    printf("same bits %d %d\n", x as byte == y, y as int8 == x)
+
+    # FIXME: these tests fail
```
Comment (Akuli) [inline tests/should_succeed/compare_signed_unsigned.jou:63]: I created #865
